### PR TITLE
Wrap more constraint violation cases to UserError

### PR DESCRIPTION
**Pull request by guangy10**
Cases covered in this PR:
 - Example inputs conflict with input constraints
 - Example inputs conflict with inline constraints
 - Suggest users to use `constrain_as_*()` when trying to export with data-dependent operations

Differential Revision: [D45666627](https://www.internalfb.com/diff/D45666627)

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_export/__init__.py [hunk 1]
```diff
@@ -101,15 +106,22 @@ def _export(
...
+                decomposition_table=DECOMP_TABLE,
+                constraints=constraints,
+                assume_static_by_default=True,
+            )
+        except (ConstraintViolationError, ValueRangeError) as e:
+            raise UserError(UserErrorType.CONSTRAIN_VIOLATION, str(e))
+        except GuardOnDataDependentSymNode as e:
+            raise UserError(
+                UserErrorType.ANTI_PATTERN,
+                f"Consider annotating your code using constrain_as_*(). {str(e)}")
```
Comment (avikchaudhuri) [inline torch/_export/__init__.py:124]: Can you please link to an exportdb example? Or do you plan to do it separately?

**Review by avikchaudhuri — APPROVED** [review]

File: torch/_export/__init__.py (hunk 1 above)
Comment (guangy10) [reply torch/_export/__init__.py:124]: Let me see if we have one for it in the export db

Comment (guangy10) [reply torch/_export/__init__.py:124]: It looks like we don't have one yet. Let's do it separately

**Comment by guangy10** [comment]
Verified the failed test works locally.

@pytorchbot merge -ic

**Pull request closed without merging**